### PR TITLE
Complete task 5 - GET /api/articles

### PR DESCRIPTION
**Pull request by Dimeben**

### CODE REVIEW

File: __tests__/articles.test.js [hunk 1]
```diff
@@ -51,3 +52,46 @@ describe("/api/articles/:article_id", () => {
       });
   });
 });
+
+describe.only("/api/articles", () => {
```
Comment (nc-verity-gregory) [inline __tests__/articles.test.js:56]: Don't leave .only in test files. This gets around the husky pre commit and now I have no confidence other tests are passing.

File: __tests__/articles.test.js [hunk 2]
```diff
@@ -51,3 +52,46 @@ describe("/api/articles/:article_id", () => {
...
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        const articles = res.body.articles;
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("comment_count", expect.any(Number));
+        });
+      });
+  });
+  test("404 - GET - sends an appropriate status and error message when an invalid URL is passed", () => {
```
Comment (nc-verity-gregory) [inline __tests__/articles.test.js:89]: You already have a test for this, this 404 is not related to this end point as it is specifically for just missing urls. This does not need to be here. You only need to test for this once and in it's own describe block.

File: app.js [hunk 3]
```diff
@@ -8,14 +8,23 @@ const {
...
+  getAllArticles,
+} = require("./controllers/articles-controller");
 
 app.get("/api/topics", getTopics);
 
 app.get("/api", getApis);
 
 app.get("/api/articles/:article_id", getArticles);
 
+app.use(express.json());
```
Comment (nc-verity-gregory) [inline app.js:22]: This is not necessary for a get request. It is also in a very weird position in your app.

File: __tests__/articles.test.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ const request = require("supertest");
 const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
 const app = require("../app");
+const jestSorted = require("jest-sorted");
```
Comment (nc-verity-gregory) [inline __tests__/articles.test.js:5]: this doesn't need to be imported, the config in package.json should be updated

File: controllers/articles-controller.js [hunk 5]
```diff
@@ -1,4 +1,7 @@
-const { selectArticles } = require("../models/articles-model");
+const {
+  selectArticles,
+  selectAllArticles,
+} = require("../models/articles-model");
 
 exports.getArticles = (req, res, next) => {
```
Comment (nc-verity-gregory) [inline controllers/articles-controller.js:6]: if you're getting a single article, the function should reflect that it is a single result not plural.

File: models/articles-model.js [hunk 6]
```diff
@@ -13,3 +13,24 @@ exports.selectArticles = (article_id) => {
...
+      ORDER BY articles.created_at DESC`
+    )
+    .then((articles) => {
+      const articlesArray = articles.rows;
+      if (articlesArray.length === 0) {
+        return Promise.reject({ status: 404, msg: "Page not found" });
+      }
+
+      for (let i = 0; i < articlesArray.length; i++) {
+        articlesArray[i].comment_count = Number(articlesArray[i].comment_count);
```
Comment (nc-verity-gregory) [inline models/articles-model.js:32]: We covered this on an NC late. No need to do this. The coalesce will make a null value 0, as you're doing a count there is no need for it. In order to get a number and not a string you can cast the count as an int and you already have an example of this that we did together.

**Review by nc-verity-gregory — COMMENTED** [review]
A good effort, please don't leave the .only in. Really I should have rejected this.

File: __tests__/articles.test.js (hunk 2 above)
Comment (Dimeben) [reply __tests__/articles.test.js:89]: Screenshot 2024-08-26 at 13 21 00

File: app.js (hunk 3 above)
Comment (Dimeben) [reply app.js:22]: Screenshot 2024-08-26 at 13 22 07
Screenshot 2024-08-26 at 13 22 25

File: __tests__/articles.test.js (hunk 4 above)
Comment (Dimeben) [reply __tests__/articles.test.js:5]: Screenshot 2024-08-26 at 13 29 17

File: controllers/articles-controller.js (hunk 5 above)
Comment (Dimeben) [reply controllers/articles-controller.js:6]: Screenshot 2024-08-26 at 13 32 59

File: models/articles-model.js (hunk 6 above)
Comment (Dimeben) [reply models/articles-model.js:32]: Screenshot 2024-08-26 at 13 37 44
